### PR TITLE
#167783502 Fetch All Tags

### PR DESCRIPTION
**Pull request by ocranbillions**
### What does this PR do?
- Fetches all tags

#### Description of Task to be completed?
Screenshot 2019-08-15 at 12 10 22 PM

    - Declare new Set
    - Fetch tagList for each article
    - Loop over each article's taglist
    - Add each tag to Set
    - Return TagsArray

#### How should this be manually tested?
- Open the review-app on this PR and make a get request to /articles/tags

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
- [#167783502](https://www.pivotaltracker.com/story/show/167783502)

#### Screenshots (if appropriate)
Screenshot 2019-08-14 at 5 05 31 PM

### CODE REVIEW

File: controllers/TagsController.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        }
+      });
+
+      let tagsArray = [];
+      articles.forEach((article) => {
+        tagsArray = tagsArray.concat(article.tagsList);
+      });
+
+      // Remove duplicates
+      const allTags = [...new Set(tagsArray)];
```
Comment (Tunji17) [inline controllers/TagsController.js]: nice work here

**Review by Tunji17 — APPROVED** [review]
LGTM

File: controllers/TagsController.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  static async getTags(req, res, next) {
+    try {
+      const articles = await Article.findAll({
+        attributes: ['tagsList'],
+        where: {
+          tagsList: { [Op.ne]: null }
+        }
+      });
+
+      let tagsArray = [];
```
Comment (profmcdan) [inline controllers/TagsController.js]:
```js
let tagSet = new Set();
articles.forEach((article) => {
       article.tagsList.map(t => {
           tagSet.add(t);
       });
});
```

**Review by profmcdan — CHANGES_REQUESTED** [review]
Implement the requested feedback

**Comment by ocranbillions** [comment]
> Implement the requested feedback

Feedback implemented

**Review by mekzy-o — APPROVED** [review]
Nice Job Samuel. LGTM

**Review by profmcdan — APPROVED** [review]

**Pull request merged**